### PR TITLE
Fix VB Spellchecker Crash

### PR DESCRIPTION
**Pull request by jmarolf**
Fixes https://github.com/dotnet/roslyn/issues/6338

### CODE REVIEW

**Comment by jmarolf** [comment]
@dotnet/roslyn-ide

**Comment by dpoeschl** [comment]
Looks good, but please add a test. :smile:

File: src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ internal abstract class AbstractSpellCheckCodeFixProvider<TSimpleName> : CodeFix
                 // Only bother with identifiers that are at least 3 characters long.
                 // We don't want to be too noisy as you're just starting to type something.
                 var nameText = name.GetFirstToken().ValueText;
-                if (nameText.Length >= 3)
+                if (nameText?.Length >= 3)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs]: When can this happen?

Comment (davkean) [inline src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs]: Add a test to hit this

Comment (jmarolf) [inline src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs]: See #6338 Any time the name doesn't exist so in `Microsoft.CodeAnalysis.` the name after the last dot does not exist and is null

**Comment by CyrusNajmabadi** [comment]
Hrmm... I'm not sure I like that.  Is it an Identifier token?  If so, I would want it to have an empty string as the ValueText.

**Comment by jmarolf** [comment]
Closed this and opened https://github.com/dotnet/roslyn/pull/6343 against stabilization

**Pull request closed without merging**